File: LoadConfig.cs
using System;
using System.IO;
using System.Collections.Generic;
using YamlDotNet.Serialization;
using YamlDotNet.Serialization.NamingConventions;

namespace LoadConfig
{
    public class DataBase
    {
        public string? Host { get; set; }
        public required int Port { get; set; }
        public string? User { get; set; }
        public string? Password { get; set; }
        public string? Name { get; set; }
    }
    public class Telegram
    {
        public int API_ID { get; set; }
        public string? API_HASH { get; set; }
        public string? Token { get; set; }
        public string? Botname { get; set; }
        public string? RequiredChannel { get; set; }
        public string? ChatId { get; set; }
    }
    public class Emby
    {
        public string? Url { get; set; }
        public string? Token { get; set; }
    }
    public class Radarr
    {
        public string? Url { get; set; }
        public string? Token { get; set; }
    }
    public class Sonarr
    {
        public string? Url { get; set; }
        public string? Token { get; set; }
        public string? AnimeUrl { get; set; }
        public string? AnimeToken { get; set; }
    }
    public class Probe
    {
        public string? Url { get; set; }
        public string? Token { get; set; }
        public string? Id { get; set; }
    }
    public class Other
    {
        public required List<int> Admins { get; set; }
        public string? OMDB { get; set; }
        public string? Ratio { get; set; }
    }
    public class AppConfig
    {
        public required DataBase DataBase { get; set; }
        public required Telegram Telegram { get; set; }
        public required Emby Emby { get; set; }
        public required Radarr Radarr { get; set; }
        public required Sonarr Sonarr { get; set; }
        public required Probe Probe { get; set; }
        public required Other Other { get; set; }
    }

    class Configure
    {
        public static void InitConfig()
        {
            string filePath = "appconfig.yaml";
            var config = LoadConfigure();
            if (!File.Exists(filePath))
            {
                Console.WriteLine("配置文件不存在, 请按照提示填写配置文件");
                config.Other = GetUserInput<Other>("其他配置");
                config.DataBase = GetUserInput<DataBase>("数据库配置");
                config.Telegram = GetUserInput<Telegram>("Telegram配置");
                config.Emby = GetUserInput<Emby>("Emby配置");
                config.Radarr = GetUserInput<Radarr>("Radarr配置");
                config.Sonarr = GetUserInput<Sonarr>("Sonarr配置");
                config.Probe = GetUserInput<Probe>("Probe配置");

                SaveConfigure(config);
            }
        }

        public static AppConfig LoadConfigure()
        {
            string filePath = "appconfig.yaml";
            if (File.Exists(filePath))
            {
                var deserializer = new DeserializerBuilder()
                    .WithNamingConvention(CamelCaseNamingConvention.Instance)
                    .Build();

                using var reader = new StreamReader(filePath);
                return deserializer.Deserialize<AppConfig>(reader);
            }
            else
            {
                return new AppConfig
                {
                    Other = new Other() { Admins = new List<int>() },
                    DataBase = new DataBase() { Port = 3066 },
                    Telegram = new Telegram(),
                    Emby = new Emby(),
                    Radarr = new Radarr(),
                    Sonarr = new Sonarr(),
                    Probe = new Probe()
                };
            }
        }

        static void SaveConfigure(AppConfig config)
        {
            string filePath = "appconfig.yaml";
            var serializer = new SerializerBuilder()
                .WithNamingConvention(CamelCaseNamingConvention.Instance)
                .Build();

            using var writer = new StreamWriter(filePath);
            serializer.Serialize(writer, config);
        }

        static T GetUserInput<T>(string configName)
        {
            Console.WriteLine($"请输入 {configName} 的配置：");

            var properties = typeof(T).GetProperties();
            var instance = Activator.CreateInstance<T>();

            foreach (var property in properties)
            {
                Console.Write($"{property.Name}: ");
                string? input = Console.ReadLine();

                if (!string.IsNullOrEmpty(input))
                {
                    try
                    {
                        object value;
                        if (property.PropertyType == typeof(string))
                        {
                            value = input;
                        }
                        else if (property.PropertyType == typeof(List<int>))
                        {
                            value = input.Split(',')
                                .Select(s => int.TryParse(s, out int value) ? value : 0)
                                .ToList();
                        }
                        else
                        {
                            var converter = System.ComponentModel.TypeDescriptor.GetConverter(property.PropertyType);
                            value = converter.ConvertFromString(input);
                        }
                        if (value != null)
                        {
                            property.SetValue(instance, Convert.ChangeType(value, property.PropertyType));
                        }
                    }
                    catch (System.Exception)
                    {
                        Console.WriteLine($"无效的输入: {input}, 无法将 {input} 转换为 {property.PropertyType}");
                    }
                }
            }

            return instance;
        }
    }
}

File: LoadConfig.py
import yaml
from dataclasses import dataclass, field
from dataclasses_json import dataclass_json
from typing import Optional, List

@dataclass_json
@dataclass
class DataBase:
    Host: Optional[str] = 'localhost'
    Port: Optional[int] = 3066
    User: Optional[str] = None
    Password: Optional[str] = None
    DatabaseName: Optional[str] = None

@dataclass_json
@dataclass
class Telegram:
    Token: Optional[str] = None
    ApiId: Optional[int] = None
    ApiHash: Optional[str] = None
    BotName: Optional[str] = None
    ChatID: Optional[int] = None
    RequiredChannel: Optional[str] = None
    NotifyChannel: Optional[str] = None

@dataclass_json
@dataclass
class Emby:
    Host: Optional[str] = None
    ApiKey: Optional[str] = None

@dataclass_json
@dataclass
class Probe:
    Host: Optional[str] = None
    Token: Optional[str] = None
    Id: Optional[str] = None

@dataclass_json
@dataclass
class Lidarr:
    Host: Optional[str] = None
    ApiKey: Optional[str] = None

@dataclass_json
@dataclass
class Radarr:
    Host: Optional[str] = None
    ApiKey: Optional[str] = None

@dataclass_json
@dataclass
class Sonarr:
    Host: Optional[str] = None
    ApiKey: Optional[str] = None

@dataclass_json
@dataclass
class SonarrAnime:
    Host: Optional[str] = None
    ApiKey: Optional[str] = None

@dataclass_json
@dataclass
class Other:
    AdminId: Optional[list] = None
    OMDBApiKey: Optional[str] = None
    Ratio: int = 1
    Wiki: Optional[str] = None

@dataclass_json
@dataclass
class Config:
    dataBase: DataBase = field(default_factory=DataBase)
    telegram: Telegram = field(default_factory=Telegram)
    emby: Emby = field(default_factory=Emby)
    probe: Probe = field(default_factory=Probe)
    lidarr: Lidarr = field(default_factory=Lidarr)
    radarr: Radarr = field(default_factory=Radarr)
    sonarr: Sonarr = field(default_factory=Sonarr)
    sonarrAnime: SonarrAnime = field(default_factory=SonarrAnime)
    other: Other = field(default_factory=Other)

def load_config():
    try:
        with open('config.yaml', 'r') as file:
            config_dict = yaml.safe_load(file)
            return Config.from_dict(config_dict) # type: ignore
    except FileNotFoundError:
        return None

def save_config(config):
    with open('config.yaml', 'w') as file:
        config_dict = config.to_dict()
        yaml.dump(config_dict, file)
    print("配置文件已保存, 当前代码存在BUG, 请手动检查配置文件是否正确")
    exit(0)

def get_user_input(cls):
    instance = cls()
    for field in cls.__dataclass_fields__.keys():
        value = input(f"请输入 {cls.__name__} 的 {field} 配置：")
        if value:
            setattr(instance, field, value)
    return instance

def init_config():
    config = load_config()
    if not isinstance(config, Config):
        print("配置文件不存在, 请按照提示填写配置文件")
        config = Config(
            other=get_user_input(Other),
            dataBase=get_user_input(DataBase),
            telegram=get_user_input(Telegram),
            emby=get_user_input(Emby),
            probe=get_user_input(Probe),
            lidarr=get_user_input(Lidarr),
            radarr=get_user_input(Radarr),
            sonarr=get_user_input(Sonarr),
            sonarrAnime=get_user_input(SonarrAnime)
        )
        save_config(config)
    else:
        return config

